src/appservice/Appservice.ts

```diff
@@ -362,12 +367,19 @@ export class Appservice extends EventEmitter {
         return this.getUserId(this.registration.sender_localpart);
     }
 
+    /**
+     * Get the application service's "bot" user device ID.
+     */
+    public get botDeviceId(): string {
```

Comment:
```suggestion
    public get botDeviceId(): string|undefined {
```

This still doesn't return just a string :)